sdk/python/feast/repo_config.py

```diff
@@ -129,11 +130,10 @@ class RegistryConfig(FeastBaseModel):
     s3_additional_kwargs: Optional[Dict[str, str]] = None
     """ Dict[str, str]: Extra arguments to pass to boto3 when writing the registry file to S3. """
 
-    sqlalchemy_config_kwargs: Dict[str, Any] = {}
-    """ Dict[str, Any]: Extra arguments to pass to SQLAlchemy.create_engine. """
-
-    cache_mode: StrictStr = "sync"
-    """ str: Cache mode type, Possible options are sync and thread(asynchronous caching using threading library)"""
+    purge_feast_metadata: StrictBool = False
```

Comment:
@tokoko Do you have any recommendation on how we should do these type of changes?  I just don't want to delete project_metadata in one go.

Reply:
from an API standpoint, since we are still below 1.0 release mark, the policy is to mark as deprecated and then remove after 2-3 minor releases.

The problem in this case is that the deletion should happen not just from API but rather there needs to be changes made in the existing registry as well. I think we will eventually need a new command for this, something like `feast migrate` (borrowing the name from airflow) which should be run after every version bump?

Reply:
I handled the case for changes to existing registry using `_sync_feast_metadata_to_projects_table` method. Something like `migrate` may help rather than embedding the change in code.